nvim/msgpack_rpc_stream.lua

```diff
@@ -80,8 +80,51 @@ function MsgpackRpcStream:read_start(request_cb, notification_cb, eof_cb)
     local pos = 1
     local len = #data
     while pos <= len do
-      type, id_or_cb, method_or_error, args_or_result, pos =
-        self._session:receive(data, pos)
+      local helper = function()
+        local a, b, c, d, e = self._session:receive(data, pos)
+        return {a, b, c, d, e}
+      end
+      local status, rv = pcall(helper)
+      if status then
+        type, id_or_cb, method_or_error, args_or_result, pos = unpack(rv)
+      else
+        -- the error message will have been put in the 2nd variable
+        local errmsg = rv
+
+        -- write the full blob of bad data to a specific file
+        local badfilename = './msgpack-fail'
+        local badfile = io.open(badfilename, 'w')
+        badfile:write(data)
+        badfile:close()
+
+        -- build a printable representation of the bad part of the string
```

Comment:
`test/helpers.lua` has a `hexdump()` function. Might be good enough to paste it into lua-client and use it here.

Reply:
ok I'll copy-paste hexdump into the file. BTW is there a more reasonable place for me to save the full data contents (instead of `./msgpack-fail`?), and is there an easy way I can turn that behaviour off in CI where we'll never be able to get at the saved data?

Reply:
I meant paste it into the project, and use the function here :)

Reply:
> is there a more reasonable place for me to save the full data contents (instead of ./msgpack-fail?), a

What's wrong with that? This situation should almost never happen. So it doesn't matter for CI or elsewhere.